code/modules/atmospherics/machinery/air_alarm/_air_alarm.dm

```diff
@@ -57,6 +57,11 @@
 	/// Used for air alarm helper called tlv_no_ckecks to remove alarm thresholds.
 	var/tlv_no_checks = FALSE
 
+	/// Used for connecting air alarm to a remote tile/zone via air sensor instead of the tile/zone of the air alarm
+	var/connected_sensor
+	/// Used to link air alarm to air sensor via map helpers
+	var/air_sensor_chamber_id
```

Comment:
Does the connected_sensor has to be weakref?

Reply:
that might be smart yea

Reply:
I've researched weakrefs a bit, it seems like it's not needed here, because I use: `RegisterSignal(connected_sensor, COMSIG_PARENT_QDELETING, PROC_REF(disconnect_sensor))`. Please clarify if I missed something, but I guess this is resolved. I added the type as you asked.

Reply:
that works too, as long as references get cleaned up people are happy